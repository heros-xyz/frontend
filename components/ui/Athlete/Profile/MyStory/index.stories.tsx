import type { Meta, StoryObj } from "@storybook/react";
import MyStory from "./index";
const meta: Meta<typeof MyStory> = {
  title: "Components/MyStory",
  component: MyStory,
  tags: ["autodocs"],
  parameters: {
    layout: "fullscreen",
  },
  argTypes: {
    gender: {
      options: ["male", "female"],
      control: { type: "select" },
    },
  },
};

export default meta;
type Story = StoryObj<typeof MyStory>;

export const MyStory1: Story = {
  args: {
    description:
      "A skilled wicketkeeper standing back, pugnacious batsman in the traditional keeping spot of No. 7 and a stout upholder of the team ethic, Matt Prior was a crucial cog in the machine that led England up the Test rankings.",
    dob: "26 feb 1982",
    gender: "male",
  },
};
