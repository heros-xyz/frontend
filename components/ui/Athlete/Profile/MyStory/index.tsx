import React, { useMemo, useState } from "react";
import {
  Box,
  Flex,
  Heading,
  Text,
  Divider,
  Center,
  Link,
} from "@chakra-ui/react";
import { If, Then } from "react-if";
import NextLink from "next/link";
import { EditIcon } from "@/components/svg/menu/EditIcon";
interface Story {
  description: string;
  dob: string;
  gender: string;
  isEdit?: boolean;
}
const MyStory: React.FC<Story> = ({ description, dob, gender, isEdit }) => {
  const [readMore, setReadMore] = useState(false);

  const descSubstring = useMemo(() => {
    return description?.substring(0, 200);
  }, [description]);

  const showReadMore = useMemo(() => {
    return description?.length > 200 && !readMore;
  }, [description, readMore]);

  return (
    <Box
      pb={5}
      bg="accent.3"
      color="primary"
      px={5}
      py={10}
      pt={5}
      borderRadius={{ xl: "12px" }}
    >
      {isEdit && (
        <Box textAlign="right">
          <Link as={NextLink} href={"/athlete/my-profile/edit-basic-info"}>
            <EditIcon cursor="pointer" />
          </Link>
        </Box>
      )}
      <Heading fontSize="xl">My Story</Heading>
      <Box fontSize={{ base: "xs", lg: "md" }} mt={2.5} wordBreak="break-word">
        <Text as="span">
          {readMore ? description : descSubstring}
          {showReadMore ? "...." : ""}
        </Text>
        <If condition={showReadMore}>
          <Then>
            <Text
              as="span"
              fontWeight={800}
              textDecoration="underline"
              ml={1}
              cursor="pointer"
              onClick={() => setReadMore(true)}
            >
              Read more
            </Text>
          </Then>
        </If>
      </Box>

      <Flex mt={7} justifyContent="space-between" alignItems="start">
        <Flex direction="column" w="40%">
          <Text fontSize="xs" fontWeight="medium">
            Date of Birth
          </Text>
          <Text fontSize="xl" textTransform={"uppercase"} fontWeight={"bold"}>
            {dob}
          </Text>
        </Flex>
        <Center height="50px">
          <Divider orientation="vertical" borderLeftColor="grey.dark" />
        </Center>
        <Flex direction="column" w="40%">
          <Text fontSize="xs" fontWeight="medium">
            Gender
          </Text>
          <Text fontSize="xl" textTransform={"uppercase"} fontWeight={"bold"}>
            {gender}
          </Text>
        </Flex>
      </Flex>
    </Box>
  );
};

export default MyStory;
