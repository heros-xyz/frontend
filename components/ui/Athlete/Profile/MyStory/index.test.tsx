import { render } from "@testing-library/react";
import MyStory from ".";

it("renders homepage unchanged", () => {
  const { container } = render(<MyStory description="" dob="" gender="" />);
  expect(container).toMatchSnapshot();
});
