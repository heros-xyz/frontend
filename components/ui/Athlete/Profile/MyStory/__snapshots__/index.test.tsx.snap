// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homepage unchanged 1`] = `
<div>
  <div
    class="css-1buonb0"
  >
    <h2
      class="chakra-heading css-6dvxm2"
    >
      My Story
    </h2>
    <div
      class="css-1qsl6ue"
    >
      <span
        class="chakra-text css-0"
      />
    </div>
    <div
      class="css-17n7yk0"
    >
      <div
        class="css-1ipuqvb"
      >
        <p
          class="chakra-text css-dikwdi"
        >
          Date of Birth
        </p>
        <p
          class="chakra-text css-1rdoige"
        />
      </div>
      <div
        class="css-1yj0ggl"
      >
        <hr
          aria-orientation="vertical"
          class="chakra-divider css-17eavvr"
        />
      </div>
      <div
        class="css-1ipuqvb"
      >
        <p
          class="chakra-text css-dikwdi"
        >
          Gender
        </p>
        <p
          class="chakra-text css-1rdoige"
        />
      </div>
    </div>
  </div>
</div>
`;
